docs/DeveloperGuide.md

```diff
@@ -1,38 +1,201 @@
+<!-- omit in toc -->
 # Developer Guide
+<!-- omit in toc -->
+## Table of contents
```

Comment:
Design & implementation appears as `Design \& implementation` in the Table of Contents.